index.js

```diff
@@ -70,13 +85,16 @@ function run(configPath, options, callback) {
         ));
     }
 
-    var maxRetries = parseInt(options.maxRetries, 10) || 0,
-        maxConcurrentWorkers = parseInt(options.maxConcurrentWorkers, 10)
-            || require('os').cpus().length,
-        workers = workerFarm({
-            maxRetries: maxRetries,
-            maxConcurrentWorkers: maxConcurrentWorkers
-        }, require.resolve('./src/webpackWorker'));
+    if (!validate(options)) {
+        return Promise.reject(new Error(
+          'Options validation failed:\n' +
+          validate.errors.map(function(error) {
+            return 'Property: "options' + error.dataPath + '" ' + error.message;
+          }).join('\n')
+        ));
+    }
+
+    var workers = workerFarm(options, require.resolve('./src/webpackWorker'));
```

Comment:
Another way is make a separate property that contain the worker-farm options object. This lead to breaking API changes, but we can keep support current API and output a depreciation warning.